### DISCUSSION TITLE
CATALOG on hive查询时报异常java.net.UnknownHostException: xxxxx

### DISCUSSION THREAD

**Discussion by zhangjjzyl** [apache/doris#32781] [A - General / Q&A]
版本：2.0.4

```
-- 创建catalogSQL
CREATE CATALOG c_hive PROPERTIES (
    'type'='hms',
    'hive.metastore.uris' = 'thrift://192.168.30.31:9083',
		'fs.defaultFS' = 'hdfs://192.168.30.31:8020',
		'hive.version' = '2.1.1');
```
可以正常show tables，按照官方文档增加了fs.defaultFS配置，查询仍然报java.net.UnknownHostException: emr-header-1
>emr-header-1是hadoop集群的host，集群未开启ha

**Comment by yagagagaga** [comment]
你需要Hadoop集群所有节点的host配置到fe/be的hosts文件里

**Reply by zhangjjzyl** [reply]
配置hosts后是可以查询的，但是如果有两个集群的hostname一样的话，就没法通过配置hosts去解决了。doris应该支持对hive返回的catalog数据进行解析，将对应的hostname替换为ip。

**Reply by zhangjjzyl** [reply]
![图片](https://github.com/apache/doris/assets/40597598/941f3508-7528-4c26-80c2-23a9e89c2c02)
我看源码里有对hive返回的数据做解析的逻辑，但是这里的逻辑是只有在返回数据的host为空时才做处理，替换为dfs.nameservices的host。这种方式也不支持非ha的集群。
